perf.groovy

```diff
@@ -37,3 +55,61 @@ def projectFolder = Utilities.getFolderName(project) + '/' + Utilities.getFolder
         }
     }
 }
+
+// Create the Linux/OSX/CentOS coreclr test leg for debug and release and each scenario
+[true, false].each { isPR ->
+    ['Ubuntu'].each { os ->
+        def osGroup = getOSGroup(os)
+        def architecture = 'x64'
+        def configuration = 'Release'
+        def newJob = job(Utilities.getFullJobName(project, "perf_${osGroup}", isPR)) {
+            steps {
+                // Set up the copies
+                // Coreclr build containing the tests and mscorlib
+                shell('curl http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/release_windows_nt/lastSuccessfulBuild/artifact/bin/tests/tests.zip -o tests.zip')
```

Comment:
Actually I don't quite agree on "we hopefully never change configurations of jobs manually". The typical step before writing a job creation script, is to manually create that job, find out and debug through each of its steps. Taking this job as the example, I actually started from copying all existing "Execute shell" steps from existing coreclr runs, which saved me a lot of time.

Across the whole task, 80% of my time was to figure out the right process to locally run coreclr perf on Linux (there wasn't any working process post RC1), and to create the manual job & validate it. The job generation part (perf.groovy changes) only took <20% of my time.

And among those 80% time to finalize the right logic, 60% of it was spent on debugging ./tests/scripts/run-xunit-perf.sh. Because this job needed to copy coreclr/corefx/test bits which were produced from master coreclr & only available on dotnet-ci server, and copyArtifact API could't copy across server boundaries, I had to create this job on dotnet-ci server, and debug there (instead of dotnet-ci2 server which my private branch was hooked up with). Because run-xunit-perf.sh had to be checked in and merged to master coreclr to take effect in the manually created job, I had to commit EVERY debugging change in that script to my private branch, then merged it to master coreclr to debug every failure in run-xunit-perf.sh. This process repeated 10+ times, which not only generated a lot of redundant merge history (debugging processes were incremental, hence I had to merge them one by one), but also wasted a lot of time on pull requests processes.

On the other hand, those "Execute shell" steps, because they do not require check-in, if I find any step is wrong, I can simply change it on the fly and then rerun the job. It doesn't cause any useless merge history in master coreclr, and is super time-efficiency as well. Of course now I abandoned copyArtifact API and use the way suggested by Matt to work around the cross CI server limitation, which resolved the cross CI server issue for this particular job. But there might be other runs in the future whose script debugging requires check-ins to master coreclr, and placing every step in a check-in script will cause pain and low efficiency in debugging.

Due to the efficiency benefit, plus it's more composable to use "Execute shell" steps rather than hiding them in a script, I actually prefer writing those small & reusable steps as Jenkins steps, and only place large & domain specific steps into scripts.